packages/web-components/fast-foundation/src/custom-properties/manager.ts

```diff
@@ -262,28 +277,64 @@ export class ConstructableStylesCustomPropertyManager extends CustomPropertyMana
  * @public
  */
 export class StyleElementCustomPropertyManager extends CustomPropertyManagerBase {
-    public readonly sheet: CSSStyleSheet;
+    private _sheet: CSSStyleSheet | null = null;
+    public get sheet(): CSSStyleSheet | null {
+        return this._sheet;
+    }
+
+    @observable
     protected customPropertyTarget: CSSStyleDeclaration;
+    private customPropertyTargetChanged(
+        prev: CSSStyleDeclaration | null,
+        next: CSSStyleDeclaration | null
+    ) {
+        if (!prev && this.queue.size) {
+            this.queue.forEach(fn => fn());
+            this.queue.clear();
+        }
+    }
+
     public readonly styles: HTMLStyleElement;
 
     constructor(style: HTMLStyleElement, client: CustomPropertyManagerClient) {
         super();
+        const controller = client.$fastController;
 
         // For HTMLStyleElements we need to attach the element
         // to the DOM prior to accessing the HTMLStyleElement.sheet
         // because the property evaluates null if disconnected
-        client.$fastController.addStyles(style);
-        this.sheet = style.sheet!;
-        this.styles = style;
-
-        this.customPropertyTarget = (this.sheet.rules[
-            this.sheet.insertRule(hostSelector)
-        ] as CSSStyleRule).style;
+        controller.addStyles(style);
 
+        this.styles = style;
         this._owner = client;
 
+        // If the element isn't connected when the manager is created, the sheet can be null.
+        // In those cases, set up notifier for when the element is connected and set up the customPropertyTarget
+        // then.
+        controller.isConnected
```

Comment:
I'm not sure if this will ever evaluate to true in the constructor...

Reply:
Maybe in this case you want to check the direct element's `isConnected` and if not connected then subscribe to the controller's `isConnected`?

Reply:
Good point, I think that should work

Reply:
on second thought, *should* `isConnected` be true in the controller ctor if the element is connected? Seems like we might want to sync that up?

Reply:
Changed to check to element's `isConnected`, let me know if I should do a follow-up PR regarding the above.

Reply:
The Controller's isConnected indicates whether it has run through its own connection logic. So, it's a bit different than the element. It will indicated whether the template and styles have been hooked up in the first render, for example.